onnxruntime/core/providers/nv_tensorrt_rtx/nv_allocator.h

```diff
@@ -56,9 +57,13 @@ class CUDAPinnedAllocator : public IAllocator {
   CUDAPinnedAllocator(const char* name, OrtDevice::DeviceId device_id)
       : IAllocator(
             OrtMemoryInfo(name, OrtAllocatorType::OrtDeviceAllocator,
-                          OrtDevice(OrtDevice::GPU, OrtDevice::MemType::HOST_ACCESSIBLE, OrtDevice::VendorIds::NVIDIA,
-                                    device_id),
-                          OrtMemTypeCPUOutput)) {}
+                          OrtDevice(OrtDevice::CPU, OrtDevice::MemType::HOST_ACCESSIBLE, OrtDevice::VendorIds::NVIDIA,
+                                    0 /*CPU device always with id 0*/),
+                          OrtMemTypeCPUOutput)) {
+    if (device_id != 0) {
+      LOGS_DEFAULT(WARNING) << "[NvTensorRTRTX EP] Device ID was non 0 for a CudaPinned allocator. The ID will be ignored.";
+    }
+  }
```

Comment:
@skottmckay is there a reason the pinned allocator was changed to be a GPU allocator ? If so I can change it back, but since it is a normal CPU allocation that does not need any changes to be used by any other EP I did set it back to CPU.

Reply:
Ok that makes sense let me revert my change. I was not aware that this is the case to be honest.